File: macos/QMK Toolbox/USB/Bootloader/USBTinyISPDevice.m
#import "USBTinyISPDevice.h"

@implementation USBTinyISPDevice

- (id)initWithUSBDevice:(USBDevice *)usbDevice {
    if (self = [super initWithUSBDevice:usbDevice]) {
        self.name = @"USBTinyISP";
        self.type = BootloaderTypeUSBTinyISP;
        self.eepromFlashable = YES;
    }
    return self;
}

-(void)flashWithMCU:(NSString *)mcu file:(NSString *)file {
    [self runProcess:@"avrdude" withArgs:@[@"-p", mcu, @"-c", @"usbtiny", @"-U", [NSString stringWithFormat:@"flash:w:%@:i", file], @"-C", @"avrdude.conf"]];
}

-(void)flashEEPROMWithMCU:(NSString *)mcu file:(NSString *)file {
    [self runProcess:@"avrdude" withArgs:@[@"-p", mcu, @"-c", @"usbtiny", @"-U", [NSString stringWithFormat:@"eeprom:w:%@:i", file], @"-C", @"avrdude.conf"]];
}

@end

File: macos/QMK Toolbox/USB/Bootloader/USBTinyISPDevice.swift
import Foundation

class USBTinyISPDevice: BootloaderDevice {
    override init(usbDevice: USBDevice) {
        super.init(usbDevice: usbDevice)
        name = "USBTinyISP"
        type = .usbTinyIsp
        eepromFlashable = true
    }

    override func flash(_ mcu: String, file: String) {
        runProcess("avrdude", args: ["-p", mcu, "-c", "usbtiny", "-U", "flash:w:\(file):i", "-C", "avrdude.conf"])
    }

    override func flashEEPROM(_ mcu: String, file: String) {
        runProcess("avrdude", args: ["-p", mcu, "-c", "usbtiny", "-U", "eeprom:w:\(file):i", "-C", "avrdude.conf"])
    }
}
